rusk-wallet/src/wallet/file_service.rs

```diff
@@ -0,0 +1,341 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::path::PathBuf;
+
+use wallet_core::Seed;
+
+use crate::crypto::decrypt;
+use crate::dat::DatFileVersion;
+use crate::Error;
+
+/// Provides access to a secure wallet file
+pub trait SecureWalletFile: Debug + Send + Sync + Clone {
```

Comment:
We have 2 abstractions now, SecureWalletFile and `WalletFilePath` I'm not a big fan of abstractions themselves and we should only keep these abstractions if a user is going to use them as a library and something needs to be implemented. IMO we can remove both of these abstractions since we only have one implementor and keep it as simple as we can

Reply:
We have discussed it with @Daksh14 and decided to leave the final decision on @ZER0